spec/lucky/memoize_spec.cr

```diff
@@ -23,11 +29,39 @@ describe "memoizations" do
     object = ObjectWithMemoizedMethods.new
 
     object.method_1.should eq "method_1"
-    2.times { object.method_1 }
+    9.times { object.method_1 }
     object.times_method_1_called.should eq 1
+  end
 
-    object.method_2.should eq 5
+  it "can cache a nil result" do
+    object = ObjectWithMemoizedMethods.new
+
+    object.method_2.should be_nil
     9.times { object.method_2 }
     object.times_method_2_called.should eq 1
   end
+
+  it "caches based on argument equality" do
+    object = ObjectWithMemoizedMethods.new
+
+    object.method_3("arg-a", "arg-b").should eq("arg-a, arg-b")
+    9.times { object.method_3("arg-a", "arg-b") }
+    object.times_method_3_called.should eq 1
+
+    object.method_3("arg-a", "arg-c").should eq("arg-a, arg-c")
+    9.times { object.method_3("arg-a", "arg-c") }
+    object.times_method_3_called.should eq 2
+
+    object.method_3("arg-a").should eq("arg-a, default-arg-b")
+    9.times { object.method_3("arg-a") }
+    object.times_method_3_called.should eq 3
+
+    object.method_3("arg-a", arg_b: "different").should eq("arg-a, different")
+    9.times { object.method_3("arg-a", arg_b: "different") }
```

Comment:
Could you add an `eq` matcher to make sure the result is the correct result when using the memoized version?